bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TableValueImpl.java

```diff
@@ -538,6 +542,14 @@ private void updateIndexKeyMappings(Long hash) {
         }
     }
 
+    // This method checks for inherent table type violation
+    private void checkInherentTypeViolation(MapValue dataMap, BTableType type) {
+        if (!TypeChecker.checkIsType(dataMap.getType(), type.getConstrainedType())) {
+            throw BallerinaErrors.createError(VALUE_INCONSISTENT_WITH_TABLE_TYPE_ERROR,
```

Comment:
Let's use InherentTypeViolation in https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/util/exceptions/BallerinaErrorReasons.java#L64 here instead.
Check map/tuple violation check